Sources/OpenAPIRuntime/Conversion/Configuration.swift

```diff
@@ -25,14 +25,45 @@ public protocol DateTranscoder: Sendable {
 }
 
 /// A transcoder for dates encoded as an ISO-8601 string (in RFC 3339 format).
-public struct ISO8601DateTranscoder: DateTranscoder {
+public struct ISO8601DateTranscoder: DateTranscoder, @unchecked Sendable {
+
+    /// The lock protecting the formatter.
+    private let lock: NSLock
+
+    /// The underlying date formatter.
+    private let locked_formatter: ISO8601DateFormatter
```

Comment:
Do we need to have a stored property for the formatter and consequently have to deal with the locking? Could we instead hold an `IOS8601DateFormatter.Options` and use this when we create the formatter? 

Alternatively, I notice that `ISO8601DateFormatter` has [`class func string(from:timeZone:formatOptions:)`](https://developer.apple.com/documentation/foundation/iso8601dateformatter/1642834-string) so maybe we just use that and pass the options that the user provides?

Reply:
Performance is a concern here, but we mostly lived with it in the past. Now that we want to allow the adopter to provide a custom `ISO8601DateFormatter` (possibly even a subclass), which isn't `Sendable`, we need to lock around it somehow.

Reply:
> Now that we want to allow the adopter to provide a custom ISO8601DateFormatter ...

I'm gently pushing back on whether that's a requirement. And whether it would be enough for them to provide just the formatter options.

Reply:
Yeah, probably. Let's see what the perf test says so that we can balance these.